framework/include/hdgkernels/DiffusionHDGAssemblyHelper.h

```diff
@@ -0,0 +1,262 @@
+
+  /**
+   * Computes a local Jacobian matrix for the weak form:
+   * (q, v) + (u, div(v))
+   * where q is the vector field representing the gradient, v are its associated test functions, and
+   * u is the scalar field
+   */
+  void vectorVolumeJacobian(const unsigned int i_offset,
+                            const unsigned int vector_j_offset,
+                            const unsigned int scalar_j_offset,
+                            const MooseArray<Real> & JxW,
+                            const QBase & qrule);
+
+  /**
+   * Computes a local residual vector for the weak form:
+   * (Dq, grad(w)) - (f, w)
+   * where D is the diffusivity, w are the test functions associated with the scalar field, and f is
+   * a forcing function
+   */
+  void scalarVolumeResidual(const unsigned int i_offset,
+                            const MooseArray<Gradient> & vector_field,
+                            const Moose::Functor<Real> & source,
+                            const MooseArray<Real> & JxW,
+                            const QBase & qrule,
+                            const Elem * const current_elem,
+                            const MooseArray<Point> & q_point);
+
+  /**
+   * Computes a local Jacobian matrix for the weak form:
+   * (Dq, grad(w)) - (f, w)
+   * where D is the diffusivity, w are the test functions associated with the scalar field, and f is
+   * a forcing function
+   */
+  void scalarVolumeJacobian(const unsigned int i_offset,
+                            const unsigned int vector_field_j_offset,
+                            const MooseArray<Real> & JxW,
+                            const QBase & qrule);
+
+  //
+  // Methods which can be leveraged both on internal sides in the kernel and by boundary conditions
+  //
+
+  /**
+   * Computes a local residual vector for the weak form:
+   * -<\hat{u}, n*v>
+   * where \hat{u} is the trace of the scalar field, n is the normal vector, and v are the test
+   * functions associated with the gradient field
+   */
+  void vectorFaceResidual(const unsigned int i_offset,
+                          const MooseArray<Number> & lm_sol,
+                          const MooseArray<Real> & JxW_face,
+                          const QBase & qrule_face,
+                          const MooseArray<Point> & normals);
+
+  /**
+   * Computes a local Jacobian matrix for the weak form:
+   * -<\hat{u}, n*v>
+   * where \hat{u} is the trace of the scalar field, n is the normal vector, and v are the test
+   * functions associated with the gradient field
+   */
+  void vectorFaceJacobian(const unsigned int i_offset,
+                          const unsigned int lm_j_offset,
+                          const MooseArray<Real> & JxW_face,
+                          const QBase & qrule_face,
+                          const MooseArray<Point> & normals);
+
+  /**
+   * Computes a local residual vector for the weak form:
+   * -<Dq*n, w> + <\tau * (u - \hat{u}) * n * n, w>
```

Comment:
where did the n * n come from ? it s not in the doco

Reply:
Fixed in doco